cmd/lakectl/cmd/store/s3.go

```diff
@@ -12,15 +13,34 @@ import (
 	"github.com/aws/aws-sdk-go/service/s3/s3iface"
 )
 
-func GetS3Client() (*s3.S3, error) {
-	sess, err := session.NewSessionWithOptions(session.Options{
-		SharedConfigState: session.SharedConfigEnable,
-	})
+func GetS3Client(intgSrc string) (*s3.S3, error) {
+	var sess *session.Session
+	var err error
+	if intgSrc == "minio" {
+		s3EndpointUrl := os.Getenv("LAKECTL_S3_ENDPOINT_URL")
+		if len(s3EndpointUrl) == 0 {
+			return nil, fmt.Errorf("LAKECTL_S3_ENDPOINT_URL environment variable is not set")
+		}
+
+		sess, err = session.NewSessionWithOptions(session.Options{
+			Config: aws.Config{
+				Endpoint:         aws.String(s3EndpointUrl),
+				Region:           aws.String("us-east-1"), // Needs region for validation as it is AWS client
+				S3ForcePathStyle: aws.Bool(true),
+			},
+			SharedConfigState: session.SharedConfigEnable,
+		})
+	} else {
+		sess, err = session.NewSessionWithOptions(session.Options{
+			SharedConfigState: session.SharedConfigEnable,
+		})
+	}
```

Comment:
1. Can we get the URL from a flag, instead, say `s3-endpoint-url`?  Every other lakectl option uses flags, this should too.
1. I would rather not have to mention "minio" at all.  If I specify `--s3-endpoint-url`, then presumably I want to use that endpoint.  It might be MinIO, it might be any other S3 compatible storage, it might even be AWS S3 just using a different endpoint.  So I would drop the flag `integration-source` entirely.

Reply:
Nice, that makes it much more clearer 👍